### ISSUE TITLE
Incorrect code generated with building library and HMR is enabled

### ISSUE THREAD

**Issue by ramkulkarni1** [webpack/webpack#2560]
When creating a library using Webpack, the code generated without HMR is as follows - 
/* 0 */
/***/ function(module, exports, __webpack_require__) {

	module.exports = __webpack_require__(1);

/***/ },

But with HMR enabled, following code is generated - 
/***/ function(module, exports, __webpack_require__) {

	__webpack_require__(1);
	__webpack_require__(2);
	module.exports = __webpack_require__(75);

/***/ },

As you can see, in the later case module.exports is assigned value returned by HMR function instead of the entry point of the library. So empty object is always returned in this case.

**Comment by bebraw** [comment]
Could you set up a minimal demonstration repo and link to that so it's possible to reproduce this?

**Comment by ramkulkarni1** [comment]
I just realized that the problem was in my configuration. This is how I had setup entry in webpack.config.js - 

entry :[
  './js/index.js',
  `webpack-dev-server/client?http://localhost:${serverPort}/`,
   'webpack/hot/dev-server',
],

This did not create any problems as long as I did not create library. But when I created the library, the last element entry was exported, which is dev-server and which does not export anything.

When I changed the order to -

entry :[
  `webpack-dev-server/client?http://localhost:${serverPort}/`,
   'webpack/hot/dev-server',
  './js/index.js',
],

exports from my library worked fine. 

So this is not a bug in webpack. Sorry for reporting a false bug.
